### PR TITLE
refactor express to handle ResponseErrors

### PR DESCRIPTION
**Pull request by andreineculau**
long road to basically remove the need for `api-handler-bootstrap.js` from atex-platform
so the core of this review is in `express/res-error.js`, otherwise it's just breaking up `express.js`

### CODE REVIEW

File: src/express/res-mixins.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  if (mediaType) {
+    this.set('content-type', mediaType);
+  }
+
+  if (!_.isUndefined(this.validate) &&
+      _.startsWith(this.get('content-type'), this.validate.schema.mediaType)
+  ) {
+    let valid = this.validate(body);
+    if (!valid) {
+      this.log.error({
```
Comment (IanSavchenko) [inline src/express/res-mixins.js]: I think it's misleading when we spit error but continue working. I would change this to `warn`.

File: src/express/middlewares.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+} from './res-error';
+
+export let bootstrap = function(fn) {
+  return asyncHandler(async function(req, res, next) {
+    bootstrapResponseError(await _.alwaysPromise(fn(req, res, next)), res);
+  });
+};
+
+export let init = function(req, res, next) {
+  res.instance = getRequestInstance(req);
```
Comment (IanSavchenko) [inline src/express/middlewares.js]: I understand that this is not part of this PR, but this line is misleading if you are out of context. We assign a property `instance` for the response which is an instance of `request` according to the right-hand value 🤔 
But in fact it's much easier: that's just a request id from AWS. So why not calling things what they are?

**Comment by andreineculau** [comment]
> The code needs restructuring,

Forgot to ask what did you mean

**Comment by IanSavchenko** [comment]
> > The code needs restructuring,
> 
> Forgot to ask what did you mean

I can explain. For example, `middlewares.js` could be broken down further to 3 separate parts: 
1. Method `middlewares.bootstrap` is actually our wrapper for all middlewares which is used in our redefined `app.use`. So the code can be moved out there since it's the only place where we use it. So it will be like 

```js
app.oldUse = app.use;
app.use = function(fn) {
    // not the best name surely
    let wrappedFn = asyncHandler(async function(req, res, next) {
      bootstrapResponseError(await _.alwaysPromise(fn(req, res, next)), res);
    });
    app.oldUse(wrappedFn);
};
```

2. Method `middlewares.init` is used like a middleware, but it's unclear from it's name what is the purpose. Why not call it `mixins-middleware` and then put it aside next to modules `mixins-req.js` and `mixins-res.js`. Would be nice.

3. Method `middlewares.xForward` is just a plain middleware by itself, not a service-wrapper-function (like `bootstrap`), not a complex meta-middleware with some extra modules (like `init/mixins`). 

So, for me all three of these methods are serving completely different purposes and should be split into separate files. 

Also, I would move out `bootstrap` from `res-error.js`, but that's not super important.

**Comment by andreineculau** [comment]
1. :clap: pushed a change (and fix along with it)
2. renamed init to applyMixins
